product_configurator_wizard/wizard/product_configurator.py

```diff
@@ -805,14 +805,17 @@ def action_config_done(self):
         if self.product_id:
             remove_cv_links = map(lambda cv: (2, cv), self.product_id.value_custom_ids.ids)
             new_cv_links = self.product_id.product_tmpl_id.encode_custom_values(custom_vals)
-            self.product_id.write({
-                'attribute_value_ids': [(6, 0, self.value_ids.ids)],
-                'value_custom_ids':  remove_cv_links + new_cv_links,
-            })
-            if self.order_line_id:
-                self.order_line_id.write(self._extra_line_values(self.order_line_id.order_id,
-                                                                 self.product_id,
-                                                                 new=False))
+
+            if self.product_id.attribute_value_ids.ids != self.value_ids.ids:
+                new_product = self.product_tmpl_id.create_variant(self.value_ids.ids, custom_vals)
+                self.order_line_id.write({'product_id': new_product.id})
```

Comment:
My biggest concern here is the assumption that two are the same if all attributes are the same, not counting custom values.  Our implementation sees custom values driving the manufacturing order, so different custom attributes do constitute a different variant.

This code alters the custom attributes on the variant, and so effectively alters every order which points to the variant.

I think this approach has merit but may already be handled better by the option introduced in #42